Data Migration Service (DMS) sdk for public preview.

hitenjava opened the pull request:
<!-- DO NOT DELETE THIS TEMPLATE -->

## Description
- Added public preview sdk for Data Migration Service (DMS) public preview release.
- The sdk is based on swagger spec here - https://github.com/Azure/azure-rest-api-specs-pr/tree/datamigration-v2017-04-15/specification/datamigration/resource-manager/Microsoft.DataMigration/2017-11-15-preview

<!--
Please add an informative description that covers that changes made by the pull request.

If you are regenerating your SDK based off of a new swagger spec, please add the link to the corresponding swagger spec pull request that has been merged in the azure-rest-api-specs repository
-->

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md).**
- [x] **The pull request does not introduce [breaking changes](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/Documentation/breaking-changes.md).**

### [General Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#general-guidelines)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#testing-guidelines)
- [x] Pull request includes test coverage for the included changes.

### [SDK Generation Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#sdk-generation-guidelines)
- [x] If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.
- [x] The generate.cmd file for the SDK has been updated with the version of AutoRest, as well as the commitid of your swagger spec or link to the swagger spec, used to generate the code.
- [x] The `*.csproj` and `AssemblyInfo.cs` files have been updated with the new version of the SDK.

hitenjava committed:
Add Data Migration Service (DMS) sdk for public preview.

hitenjava committed:
Remove ProjectArtifactsOperationsExtensions. Not required for public preview.

hitenjava committed:
Fix build.

shahabhijeet commented on src/SDKs/DataMigration/DataMigration.Tests/DataMigration.Tests.csproj:
```diff
@@ -0,0 +1,58 @@
+﻿<Project Sdk="Microsoft.NET.Sdk">
```
@hitenjava model your project looking at other projects in the repo (e.g. https://github.com/Azure/azure-sdk-for-net/blob/psSdkJson6/src/SDKs/BatchAI/BatchAI.Tests/BatchAI.Tests.csproj

hitenjava replied:
Done.

shahabhijeet commented on src/SDKs/DataMigration/DataMigration.Tests/Helpers/Utilities.cs:
```diff
@@ -0,0 +1,114 @@
+
+        private static HttpClientHandler GetHandler()
+        {
+#if DNX451
```
@hitenjava we do not support this target? These projects are no longer DNX based projects.
We only support one compile constant and that is FullNetFx

hitenjava replied:
Done.

shahabhijeet commented on src/SDKs/DataMigration/Management.DataMigration/Microsoft.Azure.Management.DataMigration.csproj:
```diff
@@ -0,0 +1,39 @@
+  
+  <!--
+
+  <ItemGroup>
```
@hitenjava please remove commented code

hitenjava replied:
Done.

shahabhijeet commented on src/SDKs/DataMigration/Management.DataMigration/Microsoft.Azure.Management.DataMigration.csproj:
```diff
@@ -0,0 +1,39 @@
+    <Description>Provides developers with libraries for the Data Migration Service (DMS) under Azure Resource manager to deploy an instance of DMS. Create, Delete, Create project, Start Migration, and more.</Description>
+    <Version>0.1.0-preview</Version>
+    <AssemblyName>Microsoft.Azure.Management.DataMigration</AssemblyName>
+    <PackageTags>Microsoft Azure resource management;Data Migration Service;REST HTTP client;azureofficial;windowsazureofficial;netcore451511</PackageTags>
```
@hitenjava add <packageReleaseNotes>
e.g. https://github.com/Azure/azure-sdk-for-net/blob/psSdkJson6/src/SDKs/BatchAI/Management.BatchAI/Microsoft.Azure.Management.BatchAI.csproj

hitenjava replied:
Done.

shahabhijeet commented on src/SDKs/DataMigration/DataMigration.Tests/ScenarioTests/CRUDProjectTests.cs:
```diff
@@ -0,0 +1,67 @@
+{
+    public class CRUDProjectTests : CRUDDMSTestsBase
+    {
+        [Fact]
```
@hitenjava use the latest resource manager 1.6.0 at least to record all your tests.

hitenjava replied:
@shahabhijeet - You mean Microsoft.Azure.ResourceManager? I only see v1.1.0-preview as the latest.

rohit-joy replied:
@shahabhijeet Where can we find the resource manager 1.6.0 to reference?

hitenjava replied:
Had a call with Abhijeet, link to resource manager - https://www.nuget.org/packages/Microsoft.Azure.Management.ResourceManager/1.6.0-preview

I will update tests to use this.

shahabhijeet reviewed (changes requested):
Tests are failing in CI

hitenjava committed:
Changes based on review feedback.

hitenjava committed:
Update models based on latest swagger spec.

hitenjava committed:
Update tests.

hitenjava committed:
Update generated models based on latest swagger.

hitenjava committed:
Update sdk based on latest swagger.

hitenjava committed:
Fix build.

hitenjava committed:
Update version

hitenjava committed:
Update tests using Azure Resource Manager 1.6.0-preview and other changes.
Changes:
- Update tests using Azure Resource Manager 1.6.0-preview and other changes.
- Update sdk models based on latest swagge in 'current' branch in 'azure-rest-spec-api-pr'
- Add _metadata folder created by running generate.cmd

hitenjava committed:
Update test

hitenjava committed:
Remove encryption for passwords as entire input is now encrypted.

shahabhijeet reviewed (approved):
Looks good, please reopen a new PR once your spec is approved and merged into public spec repo.

shahabhijeet commented:
@hitenjava closing this PR beucase you will need a separate PR that has public repo spec referenced.
